rust/core-lib/src/movement.rs

```diff
@@ -226,6 +230,37 @@ pub fn region_movement(
             }
             (offset, None)
         }
+        Movement::BeginningOfLine => {
+            let line = view.line_of_offset(text, r.end);
+            let mut bol_offset = view.offset_of_line(text, line);
+
+            let mut word_cursor = WordCursor::new(text, bol_offset);
```

Comment:
Okay I've thought about this a lot, and I think there's a simpler solution:

- move to the beginning of the line (as you have, here)
- create a WordCursor
- move to the _next_ word
- move to the _previous_ word
- ensure that you are in the correct line

and then for 'end of line' you can just flip the next/prev calls.

This also gives us another thing we can call this; it can be `MoveToLeftmostWordInLine`.

Reply:
I've actually considered this approach, and I couldn't go with it because `move to next word`  considers punctuation and special characters, so you might have to call `next word` twice instead of just simply one time.

Consider this simple line:
`let mut something = 0;`

Move to end of line, prev word then next word would have the cursor like this:
`let mut something = 0|;`